File: Escola/Cursos.Repository/Services/RecursoprovaService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  RecursoprovaService
      /// </summary>
      public partial class RecursoprovaService : BaseServices, IRecursoprovaService
      {

		public RecursoprovaService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_RECURSOPROVA
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_RECURSOPROVA> entity)
        {
            UnitOfWork.RecursoprovaRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_RECURSOPROVA
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_RECURSOPROVA entity)
        {	
            UnitOfWork.RecursoprovaRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_RECURSOPROVA
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_RECURSOPROVA entity)
        {
            UnitOfWork.RecursoprovaRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_RECURSOPROVA
        /// </summary>
        /// <returns>IEnumerable EV_RECURSOPROVA</returns>
        public IEnumerable<EV_RECURSOPROVA> GetAll()
        {
			return UnitOfWork.RecursoprovaRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_RECURSOPROVA
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_RECURSOPROVA entity)
        {
            UnitOfWork.RecursoprovaRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_RECURSOPROVA By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_RECURSOPROVA</returns>
		public EV_RECURSOPROVA GetById(int id)
        {
			return UnitOfWork.RecursoprovaRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_RECURSOPROVA> FindBy(Expression<Func<EV_RECURSOPROVA, bool>> predicate)
        {
            return UnitOfWork.RecursoprovaRepository.FindBy(predicate);
        }
      }
}

File: Escola/Cursos.Site/app/services/RecursoprovaService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 11:55:26 AM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("RecursoprovaService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Recursoprova/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Recursoprova/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Recursoprova, callback) {
            $http.post(serviceBase + "Recursoprova/post", Recursoprova)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Recursoprova, callback) {
            $http.post(serviceBase + "Recursoprova/put", Recursoprova)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Recursoprova, callback) {
            $http.post(serviceBase + "Recursoprova/delete", Recursoprova)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

